### ISSUE TITLE
Fresh install on a virtual machine fails with "Did not detect nodejs 8.x candidate for installation"

### ISSUE THREAD

**Issue by telesat158581** [bigbluebutton/bbb-install#136]
Hi.

I'm trying to install bbb on fresh ubuntu 16.04.6 LTS server on virtual machine (VMware)
Using 
wget -qO- https://ubuntu.bigbluebutton.org/bbb-install.sh | bash -s -- -v xenial-220 -a

Having a deprecation warning on nodejs 8.x and soon after that the following output (see screenshot):

![bbb-error](https://user-images.githubusercontent.com/62747153/77750559-0d7b8d80-7035-11ea-9aa1-a6366fbf33b5.jpg)

What could be the problem?

**Comment by cepster** [comment]
Looks like #103 may help.  The advice from @sandaliero seems to have gotten me over the issue.

**Comment by telesat158581** [comment]
> Looks like #103 may help. The advice from @sandaliero seems to have gotten me over the issue.

Thanks, Cepster!
I noticed and tried it out, but in my case there's no such file in /etc/apt/sources.list.d/
Besides, i have a fresh installation, and they seemed to upgrade from 2.0 to 2.2

**Comment by cepster** [comment]
Ah, too bad.  I also was going from a fresh install but their troubleshooting steps did get me past that hurdle.  Now I'm stuck with a new error trying to fetch from ubuntu.bigbluebutton.org.  But we soldier on.

Good luck!

**Comment by ffdixon** [comment]
What is the error that your seeing?

**Comment by cepster** [comment]
I don't remember what the exact error was, but it seems like it actually was network issues on my side.  Once I got those figured out everything installed perfectly.  I'm up and running.  The Node fix from @sandaliero was necessary for me.

**Comment by cepster** [comment]
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial

**Comment by ricffb** [comment]
I'm experiencing the same issue when installing on ubuntu server 16.4 hostet through Hyper-V.
The line in the script that fails seems to be this:
```shell
if ! apt-cache madison nodejs | grep -q node_8; then
      err "Did not detect nodejs 8.x candidate for installation"
fi
```
running  `apt-cache madison nodejs` returns 
```
nodejs | 4.2.6~dfsg-1ubuntu4.2 | http://gb.archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages
nodejs | 4.2.6~dfsg-1ubuntu4.2 | http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages
nodejs | 4.2.6~dfsg-1ubuntu4 | http://gb.archive.ubuntu.com/ubuntu xenial/universe amd64 Packages
```
on my machine.

**Comment by ricffb** [comment]
Manually running
```shell
curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
```
as root and then running the install script again fixed this issue for me.   
After running the script there was a file `/etc/apt/sources.list.d/nodesource.list`.  
```shell
$ cat /etc/apt/sources.list.d/nodesource.list 
deb https://deb.nodesource.com/node_8.x xenial main
deb-src https://deb.nodesource.com/node_8.x xenial main
```

**Comment by ha3an-t** [comment]
> Manually running
> 
> ```shell
> curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
> ```
> 
> as root fixed this issue for me.
> After running the command there was a file `/etc/apt/sources.list.d/nodesource.list`.
> 
> ```shell
> $ cat /etc/apt/sources.list.d/nodesource.list 
> deb https://deb.nodesource.com/node_8.x xenial main
> deb-src https://deb.nodesource.com/node_8.x xenial main
> ```
> 
> After that running the install script worked as expected.

I can't manually install nodejs too and after running that command, still there isn't such file or directory for me on /etc/apt/sources.list.d/nodesource.list

**Comment by voicerecordist** [comment]
now show this error:
```
WARNING: The following packages cannot be authenticated!
  nodejs
E: There were unauthenticated packages and -y was used without --allow-unauthenticated
```

> Manually running
> 
> ```shell
> curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
> ```
> 
> as root fixed this issue for me.
> After running the command there was a file `/etc/apt/sources.list.d/nodesource.list`.
> 
> ```shell
> $ cat /etc/apt/sources.list.d/nodesource.list 
> deb https://deb.nodesource.com/node_8.x xenial main
> deb-src https://deb.nodesource.com/node_8.x xenial main
> ```
> 
> After that running the install script worked as expected.

**Comment by ccomincini** [comment]
> s root fixed this issue for me.
> After running the command there was a file `/etc/apt/sources.list.d/nodesource.list`.

I solved the same problem with the same solution
Thanks a lot for the Hint

**Comment by lucaloiacono** [comment]
Removing the nodesource file solved the issue for me.

`rm -rf /etc/apt/sources.list.d/nodesource.list`

**Comment by ptrskowron** [comment]
This work for me too.

**Comment by boarsteas1** [comment]
This solution here did not solve the problem for me, or I got something wrong.
Anyway, I installed the correct Nodejs 16 manually, commenting out the nodejs check, but had issues with webrtc-sfu and etherpad. Then I went back and tried to fix the installation error, finding this:
https://github.com/nodesource/distributions/issues/1266#issuecomment-931600179
Everything works as expected for now. 
In my opinion a workaround of some kind should be provided in the troubleshooting section or something even if I think this problem is not on BBB's side. I am not experienced enough to propose what's the right way to go here, but...enjoy another quick and dirty workaround and keep up the good work, otherwise the installer works flawless 👍
